common/BitField.h

```diff
@@ -131,13 +131,9 @@ struct BitField
   // so that we can use this within unions
   constexpr BitField() = default;
 
-  // We explicitly delete the copy assignment operator here, because the
-  // default copy assignment would copy the full storage value, rather than
-  // just the bits relevant to this particular bit field.
-  // Ideally, we would just implement the copy assignment to copy only the
-  // relevant bits, but we're prevented from doing that because the savestate
-  // code expects that this class is trivially copyable.
-  BitField& operator=(const BitField&) = delete;
+  // Warning!  This copies ALL storage bits, not just the bits represented by this bitfield.
```

Comment:
This is a very, *very* bad idea. Accidentally using this operator due to assigning bitfield-to-bitfield is more common than you'd think, and we did run into subtle bugs from this several times in the past.

Being able to easily copy unions is not worth it - they can still be memcpy'ed around if need be.

Reply:
Being able to copy unions is required for `cgx_defaults.h` to work (and I'm not sure how feasible it is to use memcpy with that).  I wish there was an alternative (beyond adding a copy-asignment operator, and then the other constructors and such, to every single union in BPMemory.h), but my experiments found nothing that worked (making the assignment operator private kept the unions non-copyable, and marking it as deprecated didn't work if it was also default (if I recall correctly)).

Reply:
Ah, I see you only added this patch because the copy-constructor hadn't been disabled in hwtests before your sync with Dolphin's version, and existing code already relies on its presence. That's a bummer :/

Maybe a good compromise is to keep the copy assignment operator deleted, and instead only define a move assignment operator? That way, accidental copies with wrong semantics should be avoided, while allowing the convenience copy of unions to be done explicitly by using `std::move`. (Despite saying "move", the defaulted operator would still trigger a proper copy in this case. A wrapper function around `std::move` could be added under a different name to avoid any potential confusion.)

Reply:
Huh, interesting.  Adding a move assignment operator works with almost no other changes - presumably because `CGXDefault` returns the object by value instead of by reference.  It still prevents accidentally assigning bitfields with other bitfields.  There's one place in `util.cpp` where a constant reference is provided as a parameter, but it can be worked around by just assigning `hex`.